server/api.js

```diff
@@ -101,21 +101,30 @@ router.get("/quizes/search",(req, res)=>{
 	});
 });
 
-router.get("/results/search",(req, res)=>{
+router.get("/results/:id",(req, res)=>{
 	const client = getClient();
 	client.connect(()=>{
 		const db = client.db("quiz");
 		const collection = db.collection("results");
-		const quiz = { name:req.query.name };
-		collection.find(quiz).toArray((err, results)=>{
+
+		const quizId = req.params.id;
+		if (!mongodb.ObjectID.isValid(quizId)) {
+			return res.status(400).json("the 'Quiz Id' is not valid");
+		}
+		// const id = mongodb.ObjectID(quizId);
+		const quizResults = { quiz_id:quizId };
+		collection.find(quizResults).toArray((err, results)=>{
```

Comment:
Again we'll want some sort of limit/pagination stuff down the line, in case we have millions of results for a given quiz. Can be done when we do the other one though.

Reply:
Can you please explain more about this? I didn't understand how to limit it.

Reply:
I think you'll want to do something like this:
https://docs.mongodb.com/drivers/node/fundamentals/crud/read-operations/limit

Also, code-organisation-wise, all of this in-depth database stuff should probably live in `server/db.js` and this file (`server/api.js`) can just say `return getClient.fetchQuizById(id)` or something (or possibly leave in the if/else on the error handling, but delegate all the rest to the db class). It shouldn't change the functionality, but better separates out which file is in charge of what (`api.js` should do stuff like validate the input, return errors if input is invalid, return 500 if the database is down, etc.; `db.js` should be doing the guts of going into mongo and grabbing results).

(Again, not something that needs to be done now, just to bear in mind for future PRs.)